server/src/main/java/org/elasticsearch/index/mapper/NestedObjectMapper.java

```diff
@@ -212,7 +212,11 @@ public ObjectMapper merge(Mapper mergeWith, MapperService.MergeReason reason, Ma
             MapperErrors.throwNestedMappingConflictError(mergeWith.name());
         }
         NestedObjectMapper mergeWithObject = (NestedObjectMapper) mergeWith;
-        var mergeResult = MergeResult.build(this, mergeWith, reason, parentBuilderContext);
+        return merge(mergeWithObject, reason, parentBuilderContext);
+    }
+
+    ObjectMapper merge(NestedObjectMapper mergeWithObject, MapperService.MergeReason reason, MapperBuilderContext parentBuilderContext) {
```

Comment:
it is not directly visible in the PR, but these additional package private methods are now called from some tests in place of calling the public method that does the casting, no change needed in those tests.